app/models/katello/glue/candlepin/pool.rb

```diff
@@ -132,12 +132,20 @@ def import_data
         self.update_attributes(pool_attributes.except!(*exceptions))
         self.save!
         self.create_activation_key_associations
+        self.import_hosts if index_hosts
       end
 
-      def hosts
+      def import_hosts
         entitlements = Resources::Candlepin::Pool.entitlements(self.cp_id, ["consumer.uuid"])
         uuids = entitlements.map { |ent| ent["consumer"]["uuid"] }
-        ::Host.where(:id => Katello::Host::SubscriptionFacet.where(:uuid => uuids).pluck(:host_id))
+        sub_facets = Katello::Host::SubscriptionFacet.where(:uuid => uuids)
+        sub_facets.pluck(:id).each do |sub_facet_id|
+          Katello::SubscriptionFacetPool.where(:pool_id => self.id, :subscription_facet_id => sub_facet_id).first_or_create
+        end
```

Comment:
One thing i remembered, is that this won't actually delete any uneeded SubscriptionFacetPool objects.  We need to handle that here too